internal/controller/tunnelbinding_controller.go

```diff
@@ -79,7 +80,7 @@ func (r *TunnelBindingReconciler) initStruct(ctx context.Context, tunnelBinding
 	switch strings.ToLower(r.binding.TunnelRef.Kind) {
 	case "clustertunnel":
 		namespacedName = apitypes.NamespacedName{Name: r.binding.TunnelRef.Name, Namespace: r.Namespace}
-		clusterTunnel := &networkingv1alpha1.ClusterTunnel{}
+		clusterTunnel := &networkingv1alpha2.ClusterTunnel{}
```

Comment:
Maybe use var or new declaration there?

Reply:
I did not know that was best practice, thanks!

Reply:
Actually, I think I picked it up from the kubebuilder examples. Any reason why one is better than the other? I think I've used this style throughout, and would like to keep it consistent if it is just a style preference?

https://github.com/kubernetes-sigs/kubebuilder/blob/b5902cb71eb79926adb063127c4a18e19571f5d9/docs/book/src/getting-started.md?plain=1#L236
```go
        // Check if the deployment already exists, if not create a new one
	found := &appsv1.Deployment{}
	err = r.Get(ctx, types.NamespacedName{Name: memcached.Name, Namespace: memcached.Namespace}, found)
```

Reply:
If I were to do that, I'd do that for all occurrences in the repo, so let's keep that change separate from this PR.